scrapy/downloadermiddlewares/robotstxt.py

```diff
@@ -25,11 +25,21 @@ class RobotsTxtMiddleware(object):
     def __init__(self, crawler):
         if not crawler.settings.getbool('ROBOTSTXT_OBEY'):
             raise NotConfigured
-
+        self._default_useragent = crawler.settings.get('USER_AGENT', 'Scrapy')
         self.crawler = crawler
-        self._useragent = crawler.settings.get('USER_AGENT')
         self._parsers = {}
-        self._parserimpl = crawler.settings.get('ROBOTSTXT_PARSER')
+        self._parserimpl = load_object(crawler.settings.get('ROBOTSTXT_PARSER'))
+
+        # check if parser dependencies are met
+        try:
+            self._parserimpl.from_crawler(self.crawler, b'')
+        except ImportError as e:
+            # For Python 2 compatibility
+            errmsg = e.msg if hasattr(e, 'msg') else e.message
+
+            missingmodule = re.match('No module named (.+)', errmsg).group(1).strip("'")
+            raise ImportError('Unable to use \'%s\'. Do you have \'%s\' module installed?' %
+                              (self._parserimpl.__name__, missingmodule))
```

Comment:
I'd just leave it as is, to have the full original traceback :) ImportError is self-explanatory.